### ISSUE TITLE
ERROR: Cant find USB Device

### ISSUE THREAD

**Issue by AutobotZee** [onitake/daliserver#25]
Hi.
I followed the debian installation instruction and when starting the daliserver it throws me an error saying it cannot find a usb device. And I can see that my /dev has no /dali device, instead upon plugging in the device it turns up as hiddev1 in my dev/usb folder. As I am new to this can you please guide me and help me debug this error?

Error:
DALI USB multiplexer (daliserver)
Copyright (c) 2011 onitake All rights reserved.

[2021-06-25 12:40:16] INFO Starting daliserver
[2021-06-25 12:40:16] **ERROR** Can't find USB device
[2021-06-25 12:40:16] INFO Exiting

**Comment by sde1000** [comment]
You would not see a /dev/dali device unless you set up udev rules to create one, and it would just be a symlink to the hiddev device you mention. daliserver doesn't make use of this anyway, it detaches the kernel's HID driver and controls the device directly through libusb.

Can you share the output of the `lsusb` command? daliserver is looking for a device with ID 17b5:0020 — does this show up?

**Comment by AutobotZee** [comment]
HI,
Thank you for the info. I think I am getting close now. I think i have to make daliserver look for 04cc:0802. This is the Device I need to connect to. The exact lsusb output is here.

Bus 006 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 005 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 011: ID 04cc:0802 ST-Ericsson 
Bus 001 Device 003: ID 046d:c31c Logitech, Inc. Keyboard K120
Bus 001 Device 002: ID 256f:c650 3Dconnexion CadMouse
Bus 001 Device 009: ID 0b05:1867 ASUSTek Computer, Inc. AURA Custom Human interface
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

**Comment by sde1000** [comment]
No. daliserver cannot drive the Hasseb DALI interface, the USB protocol is completely different.

**Comment by TA1DB** [comment]
Does it not support any other device which is compatible to Tridonic Dali USB Adapter ( if any ? )

**Comment by sde1000** [comment]
Unfortunately there is no standard for DALI over USB, so every manufacturer does it differently. daliserver can only drive the Tridonic and Lunatone interfaces because they are actually the same device from the same manufacturer, and the author @onitake put a lot of work into reverse-engineering the protocol.

**Comment by onitake** [comment]
This is correct - I only built daliserver around the Tridonic/Lunatone DALIUSB device.
I didn't have anything else available back in 2011.

If you're looking for something more generic, please consider @sde1000 's https://github.com/sde1000/python-dali project. It also supports a few other devices. It's unlikely that I will ever add more hardware to daliserver. Pull requests are welcome, though.

**Comment by TA1DB** [comment]
> Lunatone interfaces because ... 

Lunatone can be an alternative because it is more available in the market !
